sklearn/feature_selection/tests/test_rfe.py

```diff
@@ -108,6 +108,43 @@ def test_rfe():
     assert_array_almost_equal(X_r, X_r_sparse.toarray())
 
 
+def test_rfe_sample_weights():
+    iris = load_iris()
+    X = iris.data
+    y = iris.target
+
+    clf = SVC(kernel="linear")
+    rfe = RFE(estimator=clf, n_features_to_select=1)
+
+    sample_weight_test = 2
+    class_test = 2
+
+    # Case 1 - original dataset
+    rfe.fit(X, y)
+    ranking_original = rfe.ranking_.copy()
+
+    # Case 2 - double the weight of one class's samples
+    w = np.ones(y.shape[0])
+    w[y == class_test] = sample_weight_test
+
+    rfe.fit(X, y, sample_weight=w)
+    ranking_weights = rfe.ranking_.copy()
+
+    # Case 3 - duplicate the samples of one class
+    extra_X = np.tile(X[y == class_test], (sample_weight_test - 1, 1))
+    X_duplicate = np.concatenate((X, extra_X), axis=0)
+
+    n_extra = (y == class_test).sum() * (sample_weight_test - 1)
+    extra_Y = np.ones(n_extra, dtype=int) * class_test
+    y_duplicate = np.concatenate((y, extra_Y), axis=0)
```

Comment:
The same here. We could reuse the previous indexing to select and concatenate.

Reply:
I couldn't simplify this much... The only thing I did was removing that multiplication on L137 because it was multiplying by one

```python
 n_extra = (y == class_targeted).sum()
extra_Y = np.ones(n_extra, dtype=int) * class_targeted
y_duplicate = np.concatenate((y, extra_Y), axis=0)
```